dockers/docker-dhcp-relay/base_image_files/monit_dhcp_relay.j2

```diff
@@ -0,0 +1,36 @@
+{# This template file is used to generate Monit configuration file of dhcp_relay container -#}
+
+################################################################################
+## Monit configuration file for dhcp_relay container
+## process list:
+##  dhcrelay
+################################################################################
+{# If our configuration has VLANs... #}
+{%- if VLAN_INTERFACE -%}
+ {# Count how may VLANs require a DHCP relay agent... #}
+ {%- set num_relays = namespace(count=0) -%}
+  {%- for vlan_name in VLAN_INTERFACE -%}
+    {%- if VLAN and vlan_name in VLAN and 'dhcp_servers' in VLAN[vlan_name] and VLAN[vlan_name]['dhcp_servers']|length > 0 -%}
+      {%- set num_relays.count = num_relays.count + 1 -%}
+    {%- endif -%}
+  {%- endfor -%}
+  {# if one or more VLANs require DHCP relay agent #}
+  {%- if num_relays.count > 0 -%}
+    {%- set relay_for_ipv4 = namespace(flag=False) -%}
+    {%- for vlan_name in VLAN_INTERFACE -%}
+      {%- if VLAN and vlan_name in VLAN and 'dhcp_servers' in VLAN[vlan_name] and VLAN[vlan_name]['dhcp_servers']|length >0 -%}
+        {%- for dhcp_server in VLAN[vlan_name]['dhcp_servers'] -%}
+          {%- if dhcp_server | ipv4 -%}
+            {%- set relay_for_ipv4.flag = True -%}
+          {%- endif -%}
+        {%- endfor -%}
+        {%- if relay_for_ipv4.flag -%}
+          {%- set relay_for_ipv4 = False -%}
+          {# Check the running status of each DHCP relay agent instance #}
+check program dhcp_relay|dhcrelay with path "/usr/bin/process_checker dhcp_relay /usr/sbin/dhcrelay -d -m discard -a %h:%p %P --name-alias-map-file /tmp/port-name-alias-map.txt -id {{ vlan_name }}"
```

Comment:
@jleveque At here, I used the argument `-a %h:%p P%`. However, the template file of superviord of `dhcp_relay` container which used ` -a %%h:%%p %%P`: https://github.com/Azure/sonic-buildimage/blob/ee728aab7b921ea9d0e81dda1de3a6f0279397fd/dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2#L78. In `dhcp_relay` container, the command line of `dhcrelay` process used `-a %h:%p P%`. If I used ` -a %%h:%%p %%P` in this template file, `sonic-cfggen` will generate ` -a %%h:%%p %%P` not ` -a %h:%p %P`.

Reply:
Yes, that's correct. In the supervisor config file, we need to escape the percent sign with an extra percent sign.